doc/man7/ossl-guide-quic-client-non-block.pod

```diff
@@ -0,0 +1,440 @@
+    }
+
+On a write we do not expect to see an EOF response so we treat that case in the
+same way as a fatal error.
+
+Reading a response back from the server is similar:
+
+    do {
+        /*
+         * Get up to sizeof(buf) bytes of the response. We keep reading until
+         * the server closes the connection.
+         */
+        while (!eof && !SSL_read_ex(ssl, buf, sizeof(buf), &readbytes)) {
+            switch (handle_io_failure(ssl, 0)) {
+            case 1:
+                continue; /* Retry */
+            case 0:
+                eof = 1;
+                continue;
+            case -1:
+            default:
+                printf("Failed reading remaining data\n");
+                goto end; /* Cannot retry: error */
+            }
+        }
+        /*
+         * OpenSSL does not guarantee that the returned data is a string or
+         * that it is NUL terminated so we use fwrite() to write the exact
+         * number of bytes that we read. The data could be non-printable or
+         * have NUL characters in the middle of it. For this simple example
+         * we're going to print it to stdout anyway.
+         */
+        if (!eof)
+            fwrite(buf, 1, readbytes, stdout);
+    } while (!eof);
+    /* In case the response didn't finish with a newline we add one now */
+    printf("\n");
+
+The main difference this time is that it is valid for us to receive an EOF
+response when trying to read data from the server. This will occur when the
+server closes down the connection after sending all the data in its response.
+
+In this demo we just print out all the data we've received back in the response
+from the server. We continue going around the loop until we either encounter a
+fatal error, or we receive an EOF (indicating a graceful finish).
+
+=head2 Shutting down the connection
+
+As in the QUIC blocking example we must shutdown the connection when we are
+finished with it.
+
+Even though we have received EOF on the stream that we were reading from above,
+this tell us nothing about the state of the underlying connection. Our demo
+applicaiton will initiate the connection shutdown process via
+L<SSL_shutdown(3)>.
+
+Since our application is initiating the shutdown then we might expect to see
+L<SSL_shutdown(3)> give a return value of 0, and then we should continue to call
+it until we recieve a return value of 1 (meaning we have successfully completed
+the shutdown). Since we are using a nonblocking B<SSL> object we might expect to
+have to retry this operation several times. If L<SSL_shutdown(3)> returns a
+negative result then we must call L<SSL_get_error(3)> to work out what to do
+next. We use our handle_io_failure() function that we developed earlier for
+this:
+
+    /*
+     * Repeatedly call SSL_shutdown() until the connection is fully
+     * closed.
+     */
+    while ((ret = SSL_shutdown(ssl)) != 1) {
+        if (ret < 0 && handle_io_failure(ssl, ret) == 1)
+            continue; /* Retry */
```

Comment:
I find this example problematic, since it will just hang here.

Reply:
As above.